pkg/apis/meta/v1/types.go

```diff
@@ -63,3 +63,19 @@ type SecretKeySelector struct {
 const (
 	TLSCAKey = "ca.crt"
 )
+
+// Describes how the ACME challenge self check behaves when it fails.
+// +kubebuilder:validation:Enum=WaitForever;Ignore
+type ACMEFailurePolicy string
```

Comment:
Could we rename these to something like `ACMESelfCheckFailurePolicy` to not be confused with the behavior if an ACME order failing?

Reply:
Very good idea, fixed as suggested!